src/async/imap/MCIMAPAsyncSession.cpp

```diff
@@ -338,19 +347,37 @@ IMAPAsyncConnection * IMAPAsyncSession::availableSession()
     }
 }
 
-IMAPAsyncConnection * IMAPAsyncSession::matchingSessionForFolder(String * folder)
+IMAPAsyncConnection * IMAPAsyncSession::sessionForFolderWithMinQueue(String * folder)
 {
-    for(unsigned int i = 0 ; i < mSessions->count() ; i ++) {
-        IMAPAsyncConnection * currentSession = (IMAPAsyncConnection *) mSessions->objectAtIndex(i);
-        if (currentSession->lastFolder() != NULL) {
-            if (currentSession->lastFolder()->isEqual(folder)) {
-                return currentSession;
+    IMAPAsyncConnection * chosenSession = NULL;
+    unsigned int minOperationsCount = std::numeric_limits<unsigned int>::max();
```

Comment:
Actually chosenSession being NULL is also a good indication (see availableSession).